### PR TITLE
HTCONDOR-2335 convert-string-lists

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/compat_classad.cpp [hunk 1]
```diff
@@ -297,33 +300,32 @@ bool stringListSummarize_func( const char *name,
...
-
-	sl.rewind();
-	const char *entry;
-	while ( (entry = sl.next()) ) {
+	int cnt = 0;
+	for (auto& entry : StringTokenIterator(list_str, delim_str.c_str())) {
+		cnt++;
 		double temp;
-		int r = sscanf(entry, "%lf", &temp);
+		int r = sscanf(entry.c_str(), "%lf", &temp);
```
Comment (GregThain) [inline src/condor_utils/compat_classad.cpp:307]: Not all our compilers support this yet, but at some point, we should replace these sscanf's with the double overload of `std::from_chars`, which is faster, avoids all format-string security and varags wonkiness.

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_utils/compat_classad.cpp [hunk 2]
```diff
@@ -297,33 +300,32 @@ bool stringListSummarize_func( const char *name,
...
 		if (r != 1) {
 			result.SetErrorValue();
 			return true;
 		}
-		if (strspn(entry, "+-0123456789") != strlen(entry)) {
+		if (strspn(entry.c_str(), "+-0123456789") != strlen(entry.c_str())) {
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/compat_classad.cpp]: Done

File: src/condor_utils/compat_classad.cpp [hunk 3]
```diff
@@ -218,8 +218,11 @@ bool stringListSize_func( const char * /*name*/,
 		return true;
 	}
 
-	StringList sl( list_str.c_str(), delim_str.c_str() );
-	result.SetIntegerValue( sl.number() );
+	int cnt = 0;
+	for (auto& item : StringTokenIterator(list_str, delim_str.c_str())) {
+		cnt++;
+	}
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/compat_classad.cpp]: Done

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**